### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by nathalie321** [mlflow/mlflow#11837]
### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

2.12.1

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 22.04
- **Python version**: 3.11
- **yarn version, if running the dev UI**:

### Describe the problem

log_table support for image does not display the image correctly.

![comparison](https://github.com/mlflow/mlflow/assets/153890520/c3b8d92e-1bcb-4f64-9f98-fc44b606eaf4)

### Steps to reproduce the bug

Sample from docs doesn't display image.

 image = mlflow.Image([[1, 2, 3]])
  table_dict = {
      "inputs": ["Show me a dog", "Show me a cat"],
      "outputs": [image, image],
  }
  from mlflow import MlflowClient
  df = pd.DataFrame.from_dict(table_dict)
  with mlflow.start_run() as run:
      client = MlflowClient()
      client.log_table(run.info.run_id, data=df, artifact_file="image_gen.json")

**Comment by WeichenXu123** [comment]
Hello,

The image is visible well in run details > artifacts,
but I am not sure this is bug or by design in evaluation table.

I will ask our team members and get back to you :)

**Comment by nathalie321** [comment]
Ah yes, I can clearly see it inside the artifacts.
I'm also interested in it being displayed in evaluation table, thank you!

**Comment by WeichenXu123** [comment]
@nathalie321 Yes this is a bug. Feel free to file a PR to fix it. Appreciate your reporting and contribution!

**Comment by WeichenXu123** [comment]
thanks for reporting this, I talked with our team folks, this is a bigger project than just a backend change - REACT modules are needed, we plan to fix during Q2. please stay tuned. :)

**Comment by WeichenXu123** [comment]
After code sync from our internal repository https://github.com/mlflow/mlflow/pull/12026 , this issue should be addressed.
